src/ameba/rule/typing/method_return_type_restriction.cr

```diff
@@ -0,0 +1,61 @@
+module Ameba::Rule::Typing
+  # A rule that enforces method definitions have a return type restriction.
+  #
+  # For example, these are considered valid:
+  #
+  # ```
+  # def hello : String
+  #   "hello world"
+  # end
+  #
+  # def listen(a, b) : Int32
+  #   0
+  # end
+  # ```
+  #
+  # And these are considered invalid:
+  #
+  # ```
+  # def hello
+  #   "hello world"
+  # end
+  #
+  # def listen(a, b)
+  #   0
+  # end
+  # ```
```

Comment:
Please, keep the examples minimal, functional and list the invalid cases first.

```suggestion
  # For example, this are considered invalid:
  #
  # ```
  # def hello(name = "World")
  #   "Hello #{name}"
  # end
  # ```
  #
  # And this is valid:
  #
  # ```
  # def hello(name = "World") : String
  #   "Hello #{name}"
  # end
  # ```
```